pkg/actors/targets/workflow/orchestrator/run.go

```diff
@@ -140,7 +140,15 @@ func (o *orchestrator) runWorkflow(ctx context.Context, reminder *actorapi.Remin
 			return todo.RunCompletedFalse, wferrors.NewRecoverable(todo.ErrExecutionAborted)
 		}
 	}
-	log.Debugf("Workflow actor '%s': workflow execution returned with status '%s' instanceId '%s'", o.actorID, runtimestate.RuntimeStatus(rs).String(), wi.InstanceID)
+	rs = wi.State
+
+	if setStalled(ctx, o, state, rs) {
```

Comment:
Reads a bit better in the positive `if` statement case.

```suggestion
	if isStalled(ctx, o, state, rs) {
```